crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -237,4 +237,4 @@ macro_rules! impl_exclusive_system_function {
 }
 // Note that we rely on the highest impl to be <= the highest order of the tuple impls
 // of `SystemParam` created.
-all_tuples!(impl_exclusive_system_function, 0, 16, F);
+all_tuples!(impl_exclusive_system_function, 0, 17, F);
```

Comment:
This change seems out of place here.

Reply:
Yeah, I was also divided on wether or not to include this one. One of the systems I update in the other PR to add transmission ends up needing more than 16, but since it's technically a change unrelated to transmission I wasn't sure it would make sense to keep it there or here.

I didn't include it in the change log because it was minor and mostly internal/not user-facing, but I can also add a note about it.

Reply:
IMO that system should just use a custom `SystemParam` instead. That's the idiomatic way to handle the problem, so we should use it internally instead.

Reply:
Interesting, I haven't used that yet; I'll try looking into it. I agree just bumping the limit arbitrarily is probably not a good solution, especially since next time when we add another feature we'll just have to bump it again.

Reply:
Yeah, you can also nest your parameters in tuples

```rust
fn my_system(
  (first_param, second_param): (Res<First>, ResMut<Second>),
  third_param: Query<&Third>,
  // ...
) {
  // ...
}
```
`SystemParam` is better, but if you don't want to bother with them, you can use tuples. Increasing this number generates a n² amount of code and slows down compile time disproportionately.

Reply:
Ooo, I really like this tuple-based solution, if y'all are okay with me using it instead of `SystemParams`, I'd very much prefer it, since `SystemParams` feels a bit verbose for this scenario.